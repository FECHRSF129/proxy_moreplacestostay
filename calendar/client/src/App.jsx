import React, { Component } from "react";
import axios from "axios";
import Calendar from "./Calendar.jsx";
import Guest from "./Guest.jsx";
import CalendarForm from "./CalendarForm.jsx";
import Calculator from "./Calculator.jsx";
import { BaseWrapper } from "../styled_components/Base_style.jsx";
import {
  TopInfo,
  Wrapper,
  Button,
  Form,
  ReserveButton,
  LargeText,
  MediumText,
  MediumBoldText,
  SmallText,
  CenterText,
  Label,
} from "../styled_components/App_style.jsx";
import { GuestButton, GuestInput } from "../styled_components/Guest_style.jsx";
class App extends Component {
  constructor() {
    super();
    this.state = {
      bookedDates: [],
      cleaningFee: 399,
      maxGuestCount: 8,
      pricePerNight: 550,
      rating: 0,
      reviewCount: 0,
      serviceFee: 0,
      taxes: 0,
      numberOfNights: 3,
      calendar: false,
      guest: false,
      reserve: false,
      guestCount: 1,
      guestPopUp: false,
      calendarPopUp: false,
      reservePopUp: false,
      adult: 0,
      children: 0,
      infant: 0,
    };
    this.handleClick = this.handleClick.bind(this);
    this.getPlaceInfo = this.getPlaceInfo.bind(this);
    this.addPerson = this.addPerson.bind(this);
    this.removePerson = this.removePerson.bind(this);
  }

  addPerson(e) {
    console.log("add person clicked");
    e.preventDefault();
    let key = e.target.name;
    let value = this.state[key];
    let newValue = value + 1;
    if (newValue > this.state.maxGuestCount) {
      newValue = this.state.maxGuestCount;
    }
    this.setState({
      [key]: newValue,
    });
  }
  removePerson(e) {
    e.preventDefault();
    let key = e.target.name;
    let value = this.state[key];
    let newValue = value - 1;
    if (newValue < 0) {
      newValue = 0;
    }
    this.setState({
      [key]: newValue,
    });
  }

  handleClick(e) {
    console.log("CLICKED");
    e.preventDefault();
    let key = e.target.name;
    this.setState({ [key]: !this.state[key] });
  }

  setPlaceData(data) {
    let place = data[0];
    this.setState({
      pricePerNight: place.pricePerNight,
      bookedDates: place.bookedDates,
      cleaningFee: place.cleaningFee,
      maxGuestCount: place.maxGuestCount,
      rating: place.rating,
      reviewCount: place.reviewCount,
      serviceFee: place.serviceFee,
      taxes: place.taxes,
    });
  }
  getPlaceInfo() {
    axios
      .get("/calendar/place")
      .then((response) => {
        // console.log("RESPONSE", response.data);
        this.setPlaceData(response.data);
      })
      .catch((error) => {
        // handle error
        // console.log(error);
      });
  }

  componentDidMount() {
    this.getPlaceInfo();
  }

  render() {
    return (
      <BaseWrapper>
        <Wrapper>
          <TopInfo>
            <div>
              <LargeText>${this.state.pricePerNight}/night</LargeText>
            </div>
            <div>
              <MediumText>
                {this.state.rating}({this.state.reviewCount})
              </MediumText>
            </div>
          </TopInfo>

          <CalendarForm
            name="calendar"
            onClick={this.handleClick}
            openCalendar={this.state.calendar}
          />

          <form className="guest" onClick={this.handleClick}>
            <GuestButton>
              <Label>GUESTS</Label>
              <GuestInput
                type="text"
                placeholder={this.state.guestCount}
                name="guest"
              />
            </GuestButton>
          </form>
          {this.state.guest ? (
            <Guest
              guestCount={this.state.guestCount}
              adult={this.state.adult}
              children={this.state.children}
              infant={this.state.infant}
              handleClick={this.handleClick}
              addPerson={this.addPerson}
              removePerson={this.removePerson}
            />
          ) : null}
          <ReserveButton className="reserve" onClick={this.handleClick}>
            Reserve
          </ReserveButton>
          <CenterText>You won't be charged yet</CenterText>
          <Calculator
            pricePerNight={this.state.pricePerNight}
            numberOfNights={this.state.numberOfNights}
            cleaningFee={this.state.cleaningFee}
            serviceFee={this.state.serviceFee}
            taxes={this.state.taxes}
            guestCount={this.state.guestCount}
          />
        </Wrapper>
      </BaseWrapper>
    );
  }
}

export default App;
