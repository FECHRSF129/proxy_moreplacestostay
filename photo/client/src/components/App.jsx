import React from 'react';
import axios from 'axios';
import Description from './Description.jsx';
import MainPhotos from './MainPhotos.jsx';
import LandingPage from './LandingPage.jsx';
import PropertyOverview from './PropertyOverview.jsx';
import Modal from './Modal.jsx';
import ReactCarousel from './ReactCarousel.jsx';
import ReactCarouselPhotos from './ReactCarouselPhotos.jsx';
import Share from './Share.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      properties: [],
      rooms: [],
      photos: [],
      currentPage: '',
      show: false,
      photoIndex: '',
      share: false
    }

    this.getProperties = this.getProperties.bind(this);
    this.getRooms = this.getRooms.bind(this);
    this.getPhotos = this.getPhotos.bind(this);
    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
    this.lowerPhotoIndex = this.lowerPhotoIndex.bind(this);
    this.increasePhotoIndex = this.increasePhotoIndex.bind(this);
    this.openShare = this.openShare.bind(this);
    this.hideShare = this.hideShare.bind(this);
  }

  getProperties(location) {
    axios.get(`/properties/${location}`)
    .then((results) => {
      this.setState({
        properties: results.data
      })
    })
    .catch(() => console.log('There was an error with getting the data'))
  }

  getRooms(location) {
    axios.get(`/rooms/${location}`)
    .then((results) => {
      this.setState({
        rooms: results.data
      })
    })
    .catch(() => console.log('There was an error with getting the data'))
  }

  getPhotos(location) {
    axios.get(`/photos/${location}`)
    .then((results) => {
      this.setState({
        photos: results.data
      })
    })
    .catch(() => console.log('There was an error with getting the data'))
  }

  lowerPhotoIndex() {
    this.setState({
      photoIndex: this.state.photoIndex - 1
    })
  }

  increasePhotoIndex() {
    this.setState({
      photoIndex: this.state.photoIndex + 1
    })
  }

  openModal(event) {
    console.log(event.target.getAttribute('value'));
    this.setState({
      show: true,
      photoIndex: Number(event.target.getAttribute('value'))
    });
  }

  closeModal() {
    this.setState(state => ({
      show: false
    }));
  }

  openShare() {
    this.setState({
      share: true
    })
  }

  hideShare() {
    this.setState({
      share: false
    })
  }

  componentDidMount() {
    const location = window.location.pathname.match(/(\d+)/) ? window.location.pathname.match(/(\d+)/)[0] : null;
    this.getProperties(location);
    this.getRooms(location);
    this.getPhotos(location);
    this.setState({
      show: false,
      share: false
    })
  }

  render() {
    return (
      <div>
        {this.state.share ? <Share /> : null}
        {this.state.photos.length > 0 && this.state.show === false ? <LandingPage data={this.state.properties} photos={this.state.photos} click={this.openModal} share={this.openShare} hide={this.hideShare} shareState={this.state.share}/> : null}
        {/* {this.state.photos.length > 0 && this.state.show === true ? <Modal rooms={this.state.rooms} photos={this.state.photos} closeModal={this.closeModal} show={this.state.show}/> : null} */}
        {this.state.photos.length > 0 && this.state.show === true ? <ReactCarousel rooms={this.state.rooms} photos={this.state.photos} closeModal={this.closeModal} show={this.state.show} index={this.state.photoIndex} lower={this.lowerPhotoIndex} increase={this.increasePhotoIndex}/> : null}
      </div>
    )
  }
}

export default App;