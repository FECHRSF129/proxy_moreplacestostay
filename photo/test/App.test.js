import App from '../client/src/components/App.jsx';
import Enzyme, { shallow, render, mount } from 'enzyme';

const wrapper = shallow(<App />);

describe('App Test Suite', () => {

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should render correctly with no props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should get property data during `componentDidMount()`', () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'getProperties');
    instance.componentDidMount();
    expect(instance.getProperties).toHaveBeenCalledTimes(1);
  });

  it('should get room data during `componentDidMount()`', () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'getRooms');
    instance.componentDidMount();
    expect(instance.getRooms).toHaveBeenCalledTimes(1);
  });

  it('should get photo data during `componentDidMount()`', () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'getPhotos');
    instance.componentDidMount();
    expect(instance.getPhotos).toHaveBeenCalledTimes(1);
  });

});