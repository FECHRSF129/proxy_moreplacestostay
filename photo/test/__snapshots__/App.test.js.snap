// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Suite should render correctly with no props 1`] = `<div />`;
